include/RAJA/exec-cuda/scan_cuda.hxx

```diff
@@ -72,73 +71,125 @@ namespace detail
 namespace scan
 {
 
+inline cub::CachingDeviceAllocator& getAllocator() {
+  static cub::CachingDeviceAllocator allocator(true);
+  return allocator;
+}
+
 /*!
         \brief explicit inclusive inplace scan given range, function, and
    initial value
 */
-template <typename InputIter, typename Function, typename T>
-void inclusive_inplace(const ::RAJA::cuda_exec_base&,
+template <size_t BLOCK_SIZE, bool Async,
+          typename InputIter, typename Function, typename T>
+void inclusive_inplace(const ::RAJA::cuda_exec<BLOCK_SIZE, Async>&,
```

Comment:
What's the advantage of accepting `RAJA::cuda_exec<BLOCK_SIZE, Async>` rather than `RAJA::cuda_exec_base`? I understand it will give us the block size inside the function, but that isn't used here.